# Mutation Space
The mutation space of spatially conserved amino acid residues (MSCC) is a protein structural descriptor that characterizes the frequency and chemical change for amino acids in the context of relative spatial positions. The MSCC is analogous to scoring amino acid sites in a multiple sequence alignment but with three-dimensional space. 

### Dependencies
* Python 3.0 or after
* Libraries: argparse, pathlib, openpyxl, numpy, pandas, collections, math

## Summary
Additional details on the scripts and available functions may be seen in their help messages (e.g. `prep-FASTAs.py --help`).
* `prep-FASTAs.py` - Script that constructs a file containing the FASTA sequences for crystallographically resolved residues for the user-specified protein structures and chains. The generated file is suitable for sequence alignment by other programs.
* `calc_MSCC.py` - Script that computes the MSCC for the user-specified structures and chains. Requires a Multiple Sequence Alignment file of the structures (in ClustalW format) generated by another program (see for example Clustal Omega of EMBL-EBI).
* `distances.py` - Library that stores information on the distance scoring matrices used in computing the MSCC score (e.g. Grantham distance).

## Examples


# Citation
The journal article reference for this work is:
Benjamin T. Caswell, Thomas J. Summers, Gerra Licup, and David C. Cantu. "Mutation Space of Spatially Conserved Amino Acid Sites in Proteins", *Proteins* (2022).
